src/include/OpenImageIO/memory.h

```diff
@@ -0,0 +1,80 @@
+// Copyright 2008-present Contributors to the OpenImageIO project.
+// SPDX-License-Identifier: BSD-3-Clause
+// https://github.com/OpenImageIO/oiio
+
+
+/////////////////////////////////////////////////////////////////////////////
+/// \file
+///
+/// Provides helper function to compute the memory footprint of common
+/// containers.
+///
+/////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#define OPENIMAGEIO_MEMORY_H
+
+OIIO_NAMESPACE_BEGIN
+
+/// Return the total heap allocated memory used by the container.
+/// Variant for primitive types or trivial structures.
+template<class VecType>
+inline size_t
+vector_bytes(const VecType& v)
+{
+    using T = typename VecType::value_type;
+    return v.capacity() * sizeof(T);
+}
+
+/// Return the total heap allocated memory used by the container.
+/// Variant for strings, including the reserved string entries,
+/// and each string capacity.
+template<class VecType>
+inline size_t
+vector_string_datasize(const VecType& v)
+{
+    using T = typename VecType::value_type;
+    // account for the size of elements
+    size_t size = vector_bytes(v);
+    // account for the size of allocated data
+    for (const T& ref : v)
+        size += ref.capacity();
```

Comment:
Yeah I've seen the comment in your PR afterwards, but is there anyway to test whether it is large / small at runtime ? 
Something like: `size += ref.capacity() > sizeof(std::string-2) ? 0 : ref.capacity();` ?

I have still a lot of string allocations reported by my heaptrack reports from OIIO and OSL and because I'm currently not sure if they matter or not I'd like to consider the actual. Also this is generic memory tracking for vectors of strings and it is not only used for channel names, but also for vectors of paths in the second PR.

Reply:
I know how to do it. Will post a PR that adds it to strutil.h.